### PR TITLE
add Redux to the project

### PR DESCRIPTION
**Pull request by Mo7amedSMahdi**
## Configure Redux Store and write actions and reducer for adding and removing books

- Add Redux package.
- Follow the duck pattern and add the redux file structure.
- Add books Actions creator, actions type, and Reducer.
- Add categories Actions creator, actions type, and Reducer.
- Import books and categories reducers.
- Combine books and categories reducers into root reducers.
- Create a store using the root reducer.
- Default export the store;

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+const BOOK_ADDED = 'BOOK_ADDED';
+const BOOK_REMOVED = 'BOOK_REMOVED';
+
+let lastId = 0;
```
Comment (codepantha) [inline src/redux/books/books.js]: - [x] Please follow the ducks pattern in naming your action types. It should be similar to this 👇 
`const BOOK_ADDED = bookstore/books/BOOK_ADDED`;

File: src/redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    category: book.category,
+  },
+});
+
+export const bookRemovedAction = (id) => ({
+  type: BOOK_ADDED,
+  payload: {
+    id,
+  },
+});
```
Comment (codepantha) [inline src/redux/books/books.js]: - [x] I believe, here, you wanted to say `BOOK_REMOVED`? 🤔

**Review by codepantha — CHANGES_REQUESTED** [review]
Hi @Mo7amedSMahdi 👋, 

## STATUS: REQUESTED CHANGES ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### Highlights 🎯 ⭐ 
- Meaningful commits ✔️ 
- Descriptive PR ✔️ 
- Linter checks ✔️ 
- Professional README ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 1 above)
Comment (Mo7amedSMahdi) [reply src/redux/books/books.js]: Thank you for your suggestion. I will change that.

File: src/redux/books/books.js (hunk 2 above)
Comment (Mo7amedSMahdi) [reply src/redux/books/books.js]: Sorry about that mistake. I will change it.

**Review by mikethreels — APPROVED** [review]
Hi Medo, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**